src/conn/conn_reconfig.c

```diff
@@ -44,32 +44,28 @@ __wt_conn_compat_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfi
     WT_CONFIG_ITEM cval;
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
-    uint16_t max_major, max_minor, min_major, min_minor;
-    uint16_t rel_major, rel_minor;
+    WT_VERSION new_compat, min_compat, max_compat;
     char *value;
     bool txn_active, unchg;
 
     conn = S2C(session);
     value = NULL;
-    max_major = WT_CONN_COMPAT_NONE;
-    max_minor = WT_CONN_COMPAT_NONE;
-    min_major = WT_CONN_COMPAT_NONE;
-    min_minor = WT_CONN_COMPAT_NONE;
+    new_compat = min_compat = max_compat = WT_NO_VERSION;
     unchg = false;
 
     WT_RET(__wt_config_gets(session, cfg, "compatibility.release", &cval));
     if (cval.len == 0) {
-        rel_major = WIREDTIGER_VERSION_MAJOR;
-        rel_minor = WIREDTIGER_VERSION_MINOR;
+        new_compat.major = WIREDTIGER_VERSION_MAJOR;
```

Comment:
`WIREDTIGER_VERSION_MAJOR/MINOR` are eligible to be converted into a single `WIREDTIGER_VERSION` WT_VERSION struct as in the rest of this review, but doing so ends up touching public interfaces including python and bash code.
If we want to make this change I'd prefer to split it into a separate ticket.